### PR TITLE
OSSM-3612: Add maistra-test-tool tide config and plugins

### PR DESCRIPTION
**Pull request by yxun**
This PR is the first draft of onboarding maistra/maistra-test-tool test automation. It's not related to any test run yet.
This PR only helps test contributors use Prow tide and plugins such as cherrypick, auto merge and labels when creating a PR for test cases and test scripts.

### CODE REVIEW

File: ci-operator/config/maistra/maistra-test-tool/maistra-maistra-test-tool-maistra-2.4.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+build_root:
```
Comment (jwendell) [inline ci-operator/config/maistra/maistra-test-tool/maistra-maistra-test-tool-maistra-2.4.yaml:1]: why do we have these?

I thought we were going to add only tide

Comment (yxun) [reply ci-operator/config/maistra/maistra-test-tool/maistra-maistra-test-tool-maistra-2.4.yaml]: @jwendell I removed those ci-config yamls. Yes only need tide so far.

File: core-services/prow/02_config/maistra/maistra-test-tool/_prowconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+tide:
+  queries:
+  - labels:
+    - auto-merge
```
Comment (jwendell) [inline core-services/prow/02_config/maistra/maistra-test-tool/_prowconfig.yaml:4]: we don't need this label in this section

File: core-services/prow/02_config/maistra/maistra-test-tool/_prowconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    - okay to merge
```
Comment (jwendell) [inline core-services/prow/02_config/maistra/maistra-test-tool/_prowconfig.yaml:5]: should we quote this string as in https://github.com/maistra/test-infra/blob/7de39fa91c919a4f89d09b8efe018777c83fff70/prow/config/tide.yaml#L41 ?

Comment (yxun) [reply core-services/prow/02_config/maistra/maistra-test-tool/_prowconfig.yaml:4]: @jwendell The `make update` target overwrites my quote around okay to merge . 
That's why I can't quote that string in this OpenShift CI PR.

File: core-services/prow/02_config/maistra/maistra-test-tool/_prowconfig.yaml (hunk 2 above)
Comment (yxun) [reply core-services/prow/02_config/maistra/maistra-test-tool/_prowconfig.yaml]: I removed the auto-merge label in prowconfig yaml files

**Review by jwendell — APPROVED** [review]

**Pull request merged**